src/coreclr/jit/lsra.cpp

```diff
@@ -4501,8 +4501,53 @@ void LinearScan::processBlockStartLocations(BasicBlock* currentBlock)
                 reg = REG_NEXT(reg);
             }
         }
-#endif // TARGET_ARM
     }
+#else
+    regMaskTP deadCandidates = ~liveRegs;
+    if (availableRegCount < (sizeof(regMaskTP) * 8))
+    {
+        // Mask out the bits that are between 64 ~ availableRegCount
+        // so we do not visit them in below loop.
+        regMaskTP unusedRegMask = (1ULL << availableRegCount) - 1;
+        deadCandidates &= unusedRegMask;
+    }
+
+    regMaskTP deadCandidateBit = genFindLowestBit(deadCandidates);
+    while (deadCandidates != RBM_NONE)
+    {
+        deadCandidateBit         = genFindLowestBit(deadCandidates);
+        regNumber  reg           = genRegNumFromMask(deadCandidateBit);
+        RegRecord* physRegRecord = getRegisterRecord(reg);
+        deadCandidates &= ~deadCandidateBit;
```

Comment:
`deadCandidates ^= deadCandidateBit` is cheaper since we know `deadCandidateBit` is exactly 1-bit and that its set so therefore will toggle.